### PR TITLE
feat: Add slug for post url (#90)

### PR DESCRIPTION
**Pull request by ZajoncM**
Add slug for post url

### CODE REVIEW

File: frontend/src/components/DisplayPost.js [hunk 1]
```diff
@@ -1,19 +1,25 @@
 import React, { useState, useEffect } from 'react'
 import { fetchSinglePost } from '../utils'
-import { useParams } from 'react-router-dom'
 import Post from '../components/Post'
 import PostBanner from './PostBanner'
+import useParamsWithoutSlug from '../utils/customHooks/useParamsWithoutSlug'
+import { useHistory } from 'react-router'
```
Comment (Darex1991) [inline frontend/src/components/DisplayPost.js:6]: small thing, but please move external imports to the top 

```
...some external

...some internal
```

**Review by Darex1991 — COMMENTED** [review]
👍 it will be nice to see some tests :)

File: frontend/src/utils/customHooks/useParamsWithoutSlug.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { useParams } from 'react-router-dom'
+
+const useParamsWithoutSlug = () => {
+  const { id } = useParams()
+
+  const dashIndex = id.indexOf('-')
+
+  return dashIndex !== -1
```
Comment (LukeP91) [inline frontend/src/utils/customHooks/useParamsWithoutSlug.js:8]: I'm not sure if it's common in React of even JS, but IMO a ternary operator that doesn't easily fit in one line should be changed to if statement.

**Review by qbart — APPROVED** [review]

File: frontend/src/__tests__/components/DisplayPost.test.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          </Switch>
+        </Router>
+      </Provider>
+    ),
+    history,
+  }
+}
+
+describe('DisplayPost', () => {
+  it.only('sets adds the post slug in url when entering post details without providing slug', async () => {
```
Comment (Darex1991) [inline frontend/src/__tests__/components/DisplayPost.test.js:38]: please remove `only` :)

File: frontend/src/utils/customHooks/useParamsWithoutSlug.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { useParams } from 'react-router-dom'
+
+const useParamsWithoutSlug = () => {
+  const { id } = useParams()
+  const dashIndex = id.indexOf('-')
+
+  if (dashIndex === -1) {
```
Comment (simon2k) [inline frontend/src/utils/customHooks/useParamsWithoutSlug.js:7]: @ZajoncM replace this comparison logic with a simple: `id.includes('-')`

File: frontend/src/tests/test-server.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+// src/mocks/server.js
+import { setupServer } from 'msw/node'
+import { handlers } from './handlers'
+
+// This configures a request mocking server with the given request handlers.
```
Comment (simon2k) [inline frontend/src/tests/test-server.js:5]: @ZajoncM curious, are these comments auto-generated or added by you? In both cases, we can strip them as they do not convey any additional knowledge to the solution

File: frontend/src/utils/customHooks/useParamsWithoutSlug.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    return { id, slug: false }
+  }
+  return { id: id.slice(0, dashIndex), slug: id.slice(dashIndex + 1) }
```
Comment (simon2k) [inline frontend/src/utils/customHooks/useParamsWithoutSlug.js:10]: If the execution gets here, split the `id` by `-`

**Pull request merged**